packages/components/src/dropdown-menu/index.js

```diff
@@ -35,14 +35,18 @@ function DropdownMenu( {
 	className,
 	controls,
 	hasArrowIndicator = false,
+	showLabel = false,
 	icon = 'menu',
 	label,
 	popoverProps,
 	toggleProps,
+	onToggle,
 	menuProps,
 	// The following props exist for backward compatibility.
 	menuLabel,
 	position,
+	helperUI,
+	dropDownRef,
```

Comment:
I'd like to resurface this. I feel the DropdownMenu changes are all unnecessary because they are all here to serve the purpose of one MenuItems (the link one), for this reason, I think we probably should not touch DropdownMenu and Toolbar and just handle this logic when defining the MenuItem.

Reply:
The reason for all this is ARIA. Initially I've hijacked a menu item, but then @afercia [explained](https://github.com/WordPress/gutenberg/pull/16200#issuecomment-535066126) that:

<< ARIA menus can contain exclusively menuitems, menuitemcheckboxes, menutiemradios (and separator). The form within the menu breaks the ARIA pattern: the interaction isn't the expected one and screen readers may announce the menu improperly.>>

So I found this solution to have a helperUI prop that can add stuff in the menu which is NOT a menu item. dropDownRef is needed for selectively keeping the menu open, also for a11y reasons.

Reply:
I still think we shouldn't hack the DropdownMenu component to achieve this. What if the URL input is not the last item in the menu? it will break right? This make me thing, maybe this is not a DropdownMenu at all.

Reply:
> maybe this is not a DropdownMenu
Ok I'll try this approach ;)